pkg/bitcoin/hd.go

```diff
@@ -64,3 +70,46 @@ func (s *Service) DeriveExtendedKey(
 	response.ChainCode = xKey.ChainCode()
 	return response, nil
 }
+
+// Useful service to get keypair (xpub + privKey) from a seed for testing.
+// Random seed is generated if no seed is provided.
+func (s *Service) GetKeypair(seed string, chainParams ChainParams) (Keypair, error) {
+	var (
+		seedBytes []byte
+		response  Keypair
+	)
+
+	if seed == "" {
+		// Generate a random seed at the recommended length.
+		generatedSeed, err := hdkeychain.GenerateSeed(hdkeychain.RecommendedSeedLen)
+		if err != nil {
+			return response, err
+		}
+
+		seedBytes = generatedSeed
+
+	} else {
+		seedBytes = []byte(seed)
+	}
+
+	// Generate a new master node using the seed.
+	masterKey, err := hdkeychain.NewMaster(seedBytes, chainParams)
+	if err != nil {
+		return response, err
+	}
+
+	// Derive the extended key for account 0.
+	// This gives the path: m/0H
+	accountKey, err := masterKey.Derive(hdkeychain.HardenedKeyStart + 0)
```

Comment:
This is not following BIP44 standard, so you'll have unexpected results. Since you already have the master key here, you can simply use the `DeriveExtendedKey` service method.